test/e2e/push_test.go

```diff
@@ -63,4 +63,127 @@ var _ = Describe("Podman push", func() {
 		push.WaitWithDefaultTimeout()
 		Expect(push.ExitCode()).To(Equal(0))
 	})
+
+	It("podman push to local registry with authorization", func() {
+		authPath := filepath.Join(podmanTest.TempDir, "auth")
+		os.Mkdir(authPath, os.ModePerm)
+		os.MkdirAll("/etc/containers/certs.d/localhost:5000", os.ModePerm)
+		debug := podmanTest.SystemExec("ls", []string{"-l", podmanTest.TempDir})
+		debug.WaitWithDefaultTimeout()
+
+		cwd, _ := os.Getwd()
+		certPath := filepath.Join(cwd, "../", "certs")
+
+		setup := podmanTest.SystemExec("cp", []string{filepath.Join(certPath, "domain.crt"), "/etc/containers/certs.d/localhost:5000/ca.crt"})
+		setup.WaitWithDefaultTimeout()
+
+		setup = podmanTest.SystemExec("getenforce", []string{})
+		setup.WaitWithDefaultTimeout()
+		if setup.OutputToString() == "Enforcing" {
+
+			setup = podmanTest.SystemExec("setenforce", []string{"0"})
+			setup.WaitWithDefaultTimeout()
+
+			defer podmanTest.SystemExec("setenforce", []string{"1"})
+		}
+
+		session := podmanTest.Podman([]string{"run", "--entrypoint", "htpasswd", "registry:2", "-Bbn", "podmantest", "test"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
+		f, _ := os.Create(filepath.Join(authPath, "htpasswd"))
+		defer f.Close()
+
+		f.WriteString(session.OutputToString())
+		f.Sync()
+		debug = podmanTest.SystemExec("cat", []string{filepath.Join(authPath, "htpasswd")})
+		debug.WaitWithDefaultTimeout()
+
+		session = podmanTest.Podman([]string{"run", "-d", "-p", "5000:5000", "--name", "registry", "-v",
+			strings.Join([]string{authPath, "/auth"}, ":"), "-e", "REGISTRY_AUTH=htpasswd", "-e",
+			"REGISTRY_AUTH_HTPASSWD_REALM=Registry Realm", "-e", "REGISTRY_AUTH_HTPASSWD_PATH=/auth/htpasswd",
+			"-v", strings.Join([]string{certPath, "/certs"}, ":"), "-e", "REGISTRY_HTTP_TLS_CERTIFICATE=/certs/domain.crt",
+			"-e", "REGISTRY_HTTP_TLS_KEY=/certs/domain.key", "registry:2"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
+		session = podmanTest.Podman([]string{"logs", "registry"})
+
+		login := podmanTest.Podman([]string{"login", "-u", "podmantest", "-p", "test", "localhost:5000"})
+		login.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
+		push := podmanTest.Podman([]string{"push", ALPINE, "localhost:5000/my-alpine"})
+		push.WaitWithDefaultTimeout()
+		Expect(push.ExitCode()).To(Equal(0))
+	})
+
+	It("podman push to docker-archive", func() {
+		session := podmanTest.Podman([]string{"push", ALPINE, "docker-archive:/tmp/alp:latest"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+		clean := podmanTest.SystemExec("rm", []string{"/tmp/alp"})
+		clean.WaitWithDefaultTimeout()
+		Expect(clean.ExitCode()).To(Equal(0))
+	})
+
+	It("podman push to docker daemon", func() {
+		setup := podmanTest.SystemExec("bash", []string{"-c", "systemctl status docker 2>&1"})
```

Comment:
Can you wrap a conditional around this one to check for the presence of docker and skip otherwise?  Not every test machine will continue to have docker on it, esp when we are testing "inside" a podman run image.

Reply:
Tests running inside container already considered. In following lines, there is an "else if" check for the command exit without 0, and if will skip this test if exit code is not 0 and output is not include "Active: inactive".